### DISCUSSION TITLE
Tower Defense - Code refactoring

### DISCUSSION THREAD

**Discussion by TimWaeckerle** [Freaky2164/TowerDefense#37] [General]
Hi all, 

We refactored our code to match the "Clean Code" guidelines.
Feel free to have a look at our new document which includes our changes.

 [Clean Code](https://github.com/argastle/TowerDefense/blob/main/Projektmanagement/Clean%20Code/Code%20Refactoring%20Summary.md)

Greetings 
TD Team

**Comment by eileen522** [comment]
Hi Tower Defense Team,

Great job on refactoring your code to adhere to the "Clean Code" guidelines! It's great to see that you have taken steps to improve the quality and maintainability of your code.

I like that you went into detail on the changes you made. It was quite interesting to learn about the naming conventions in C#. I've seen the use of prefixing private variables and functions with an underscore in Dart code before, so maybe that's where the practice originated from.

Let's hope that all the hard work you put into this refactoring pays off in terms of increased efficiency and quality of your code. We are definitely looking forward to reading more updates from you in the future. Keep up the good work!

Kind regards,
The team behind eiBoard

**Reply by WeberWr** [reply]
Hey eiBoard,

thanks for your effort for checking out our blog.
We think we also need to do more refactoring to really match the rules of clean code, but a First Step is made.

We try the best to give you interesting updates every week. You will see more of game in the few weeks.

Regards,
TD

**Comment by Mohmad-Naser-alnakeshbandi** [comment]
Hello Tower Defense Team,

I wanted to drop a quick note to express my appreciation for your efforts in improving your codebase to conform with the "Clean Code" guidelines. It's always encouraging to see teams take proactive steps to enhance the quality and maintainability of their code.

I found it fascinating that you shared the specifics of the changes you made, particularly the C# naming conventions. It's intriguing to see how different programming languages have their own unique practices, and how these practices can influence one another.

I also just talk with one of your team members about the possibility of a "dark" mode, I am waiting for it ;)

I'm optimistic that your hard work in refactoring will pay off in terms of improved efficiency and overall code quality. We're eagerly anticipating future updates from your team. Keep up the excellent work!

Best regards,
Ghalia Team

**Reply by WeberWr** [reply]
Hey Ghalia,
thanks for your comment, we also believe that the refactoring will help us to be more efficient. 

We also talked about a Dark Mode before, but we not really now what that means, because we do not really have really bright colors, so there is no real effect a Dark Mode would have.

Thanks for your Effort
TD

> 1. I give you the hint to be here the next blogs, maybe you can play the game in the near future
